### PR TITLE
[CRM][Frontend] - Test Section in Task Template Editor

### LINKED ISSUE #215: [CRM][Frontend] - Test Section in Task Template Editor
### Description

Within the editor, the test section should look like the following:

Image

The tests should be displayed in the blue section of the scaffolded page:

Image

The test section should have 3 tabs, one for all test cases, private test cases, and public test cases. The implementation should create a `TestCard` component, that takes in a test as its parameters and displays information, allowing expected inputs and outputs to be edited. The list of test cards should also be scrollable.

The "Add Test" button should drop the list down and insert an empty test card that can have its fields filled. If there is no input or output added, it should error the user (but this might need to be done in the submission PR). 

Locking and unlocking, duplicating, and deleting should all be supported on the client side.

### Acceptance Criteria

- Implementation matches requirements above

### PR DESCRIPTION
**Pull request by LOTaher**
# [CRM][Frontend] - Test Section in Task Template Editor

## Changes

- Added test section in the task editor
- All state (creating, duplicating, deleting, editing, etc) is all handled on the client side for now
- State is maintained between tabs

---

## Notes

- Right now, the title and add test button are sticky. Unsure what design wanted, still waiting on a response

---

## Screenshots

image

image

---

## Checklist

Please go through all items before requesting reviewers:

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline warnings
- [x] All code follows repository-configured formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots included for UI changes
- [x] Remove non-applicable sections of this template
- [x] PR assigned to yourself
- [x] Reviewers requested & Slack ping sent
- [x] PR linked to the issue (fill in 'Closes #')
- [x] If design-related, notify the designer in Slack

---

## Closes

Closes #215

### CODE REVIEW

File: src/lib/components/core/TestCard.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                <div className="flex items-center gap-2">
+                    <button
+                        className="p-1 text-gray-400 hover:cursor-pointer hover:text-gray-600"
+                        onClick={(e) => {
+                            e.stopPropagation();
+                            onToggle();
+                        }}
+                    >
+                        {isPrivate ? (
+                            <Lock width={16} height={16} />
```
Comment (cherman23) [inline src/lib/components/core/TestCard.tsx]: These and the other lucide-icons should use the tailwind classes. This would be `className: size-4` to be consistent with other pages.

**Review by cherman23 — APPROVED** [review]
